src/stores/ExploreDataStore.ts

```diff
@@ -0,0 +1,130 @@
+import { makeAutoObservable } from "mobx";
+import { ConstraintInstance, TypeFormatTuple, WorkflowConfig, WorkflowSolution } from "./WorkflowTypes";
+import { makePersistable } from "mobx-persist-store";
+
+const emptyWorkflowConfig = () => {
+  return {
+    domain: undefined,
+    inputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple ],
+    outputs: [ [{id:"",label:""}, {id:"",label:""}] as TypeFormatTuple],
+    constraints: [ {constraint: {id:"",label:""}} as ConstraintInstance ],
+    minSteps: 3,
+    maxSteps: 4,
+    timeout: 120,
+    solutionCount: 10,
+    run_id: ""
+  }
+}
+
+export class ExploreDataStore {
+
+  workflowConfig: WorkflowConfig = emptyWorkflowConfig();
+  workflowSolutions: WorkflowSolution[] = [];
+  isGenerating: boolean = false;
+  generationError: string = "";
+
+  constructor() {
+    makeAutoObservable(this, {}, { deep: true });
+    makePersistable(this, { 
+      name: "ExploreDataStore",
+      properties: ["workflowConfig", "workflowSolutions"],
+      storage: window.localStorage
+    });
+  }
+
+  inputsOutputsToJSON(values: TypeFormatTuple[], dataRoot: string, formatRoot: string) {
+    return values.filter(value => value[0] !== undefined && value[1] !== undefined && value[0]!.id !== "" && value[1]!.id !== "")
```

Comment:
We should at some point add the field validation, where only valid (non-empty) inputs are obligatory.

Reply:
Agreed, that would be a better approach indeed